docs/mcd.md

```diff
@@ -0,0 +1,326 @@
+
+To avoid confusion, in the following we use the term "field" in place of "connection attribute," and use the generic term "attribute" to mean an XML element attribute. 
+
+## XML Elements
+
+### `<connection-fields>`
+
+This is the parent element for fields.
+
+### `<field>`
+
+Each connection attribute is represented by a field element in the XML. The field element has the following XML attributes. 
+
+| Name  | Meaning | Optional? | Value Notes | Other Notes |
+| ----  | ------- | --------- | ----------- | ----------- |
+| name  | Unique name of the attribute, which is used in the ConnectionBuilder() | No | Names must be unique, and cannot be any of the reserved names `dbname`, `schema`, `tablename`. | If there is a Tableau-defined name for this attribute, that name must be used. See [Tableau-defined attribute names](https://gitlab.tableausoftware.com/connectivity/connector-plugin-sdk-internal/blob/master/docs/attributes.md). <br> **TODO correct link** |
+| label | Label that appears on the connection dialog for the field | No | | |
+| value-type | Dictates the default validation rule and the UI widget | No | Allowed Values / UI Widget Type <br> <ul><li>`string` / text box</li><li>`option` / drop-down</li><li>`boolean` / checkbox</li><li>`file` / file picker</li> | In the 2020.2 release `file` is not supported. |
+| default-value | Default value for the attribute | Yes	| Default values by value-type <br> <ul><li>string: `""`</li><li>option: first option</li><li>boolean: `false`</li><li>file: `""`</li></ul>| |
+| optional | Whether the user must specify a value for the attribute | Yes | Allowed values: `true`, `false`. <br> Default value: `false`. | |
+| editable | Whether the user can edit the attribute | Yes | Allowed values: `true`, `false`. <br> Default value: `true`. | When set to `false`, the attribute is not shown in the connection dialog, and its default-value is passed to the ConnectionBuilder(). |
+| secure | Whether the attribute value is sensitive data, and should be suppressed from logs | Yes | Allowed values: `true`, `false`. <br> Default value: `false`. | In the 2020.2 release only `password` is allowed to be secure. The connector will not load if other fields are specified as secure. |
+| category | Specifies which tab contains the field for the attribute. | Yes | Allowed values: <br> <ul><li>`endpoint` (for server, port, etc.)</li><li>`metadata` (for data hierarchy)</li><li>`authentication`</li><li>`general`</li><li>`initial-sql`</li><li>`advanced`</li></ul> Default value: `general` | In the 2020.2 release this has minimal effect; `<initial-sql>` and `<advanced>` are not supported. | 
+
+### `<validation-rule>`
+
+An optional child of field, validation-rule lets you specify a regular expression for validating the user input. 
+It has the following XML attributes.
+
```

Comment:
Add note that the validation rule is not enforced in current alpha version.

Reply:
thanks